great_expectations/rule_based_profiler/util.py

```diff
@@ -242,36 +242,80 @@ def compute_quantiles(
         metric_values,
         q=(false_positive_rate / 2),
         axis=0,
-        interpolation="linear",  # can be omitted ("linear" is default)
     )
     upper_quantile = np.quantile(
         metric_values,
         q=1.0 - (false_positive_rate / 2),
         axis=0,
-        interpolation="linear",  # can be omitted ("linear" is default)
     )
     return lower_quantile, upper_quantile
 
 
-def compute_bootstrap_quantiles(
+def _compute_bootstrap_quantiles_point_estimate_legacy(
     metric_values: np.ndarray,
     false_positive_rate: np.float64,
     n_resamples: int,
 ) -> Tuple[Number, Number]:
     """
-    Internal implementation of the "bootstrap" estimator method, returning confidence interval for a distribution.
-    See https://en.wikipedia.org/wiki/Bootstrapping_(statistics) for an introduction to "bootstrapping" in statistics.
+    An internal implementation of the "bootstrap" estimator method, returning a point estimate for a population
+    parameter of interest (lower and upper quantiles in this case). See
+    https://en.wikipedia.org/wiki/Bootstrapping_(statistics) for an introduction to "bootstrapping" in statistics.
+
+    This implementation has been replaced by "compute_bootstrap_quantiles_bias_corrected_point_estimate" and only
+    remains to demonstrate the performance improvement achieved by correcting for bias. Upon the implementation of a
+    Machine Learning Lifecycle framework, the performance improvement can be documented and this legacy method can be
+    removed from the codebase.
+    """
+    bootstraps: np.ndarray = np.random.choice(
+        metric_values, size=(n_resamples, metric_values.size)
+    )
+    lower_quantiles: Union[np.nd_array, Number] = np.quantile(
+        bootstraps,
+        q=false_positive_rate / 2,
+        axis=1,
+    )
+    lower_quantile_point_estimate: Number = np.mean(lower_quantiles)
+    upper_quantiles: Union[np.nd_array, Number] = np.quantile(
+        bootstraps,
+        q=1.0 - (false_positive_rate / 2),
+        axis=1,
+    )
+    upper_quantile_point_estimate: Number = np.mean(upper_quantiles)
+    return lower_quantile_point_estimate, upper_quantile_point_estimate
 
-    This implementation is sub-par compared to the one available from the "SciPy" standard library
-    ("https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.bootstrap.html"), because it introduces bias and
-    does not handle multi-dimensional statistics (unlike "scipy.stats.bootstrap", which corrects for bias and is
-    vectorized, thus having the ability to accept a multi-dimensional statistic function and process all dimensions).
 
-    This implementation will be replaced by "scipy.stats.bootstrap" when Great Expectations can be upgraded to use a
-    more up-to-date version of the "scipy" Python package (the currently used version does not have "bootstrap").
+def compute_bootstrap_quantiles_bias_corrected_point_estimate(
```

Comment:
:heart: Thank you @NathanFarmer for digging into this. I think this also sets the bar for most-extensively researched and best-supported DocString we have in our codebase (by far)